### PR TITLE
Behavior test for SolrSearch, plus placeholder

### PR DESCRIPTION
**Pull request by kenprice**
This includes a phpspec test describing the basic behavior of SolrSearch.

### CODE REVIEW

File: src/builders/query-builder.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+namespace SearchApi\Builders;
+
+/**
+ * Interface QueryBuilder - Responsible for building a query string for search providers.
+ *
+ * @method string build_query (SearchTerm[]|null $keywords, SearchApi\Models\SearchOptions|null)
+ *   Build a query string from provided search terms and search options
+ */
+interface QueryBuilder {
+  function build_query( $keywords, $options = null );
```
Comment (idmontie) [inline src/builders/query-builder.php]: Builders should just expose a `build` method to maintain consistency.

File: src/builders/solr-query-builder.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+use SearchApi\Builders\QueryBuilder;
+
+/**
+ * Class SolrQueryBuilder - Responsible for building a query string for SolrSearch.
+ *
+ * @method string build_query (SearchTerm[]|null $keywords, SearchApi\Models\SearchOptions|null)
+ *   Build a query string from provided search terms and search options
+ */
+class SolrQueryBuilder implements QueryBuilder {
+  function build_query( $keywords, $options = null ) {
```
Comment (idmontie) [inline src/builders/solr-query-builder.php]: The Interface will change, so this should be just `build`.

Comment (kenprice) [inline src/builders/solr-query-builder.php]: @idmontie Fixed in 5431531

**Comment by idmontie** [comment]
:+1: 
:zap:

**Comment by idmontie** [comment]
:100: / :100:

**Pull request merged**